stackslib/src/chainstate/nakamoto/staging_blocks.rs

```diff
@@ -302,6 +302,44 @@ impl<'a> NakamotoStagingBlocksConnRef<'a> {
             .optional()?)
     }
 
+    /// Get the block ID of a staging block from block hash.
+    /// Used for downloads
+    pub fn get_block_id_by_block_hash(
+        &self,
+        block_hash: &BlockHeaderHash,
+    ) -> Result<Option<StacksBlockId>, ChainstateError> {
+        let sql = "SELECT index_block_hash FROM nakamoto_staging_blocks WHERE block_hash = ?1";
+        let args = params![block_hash];
+
+        let mut stmt = self.deref().prepare(sql)?;
+        Ok(stmt
+            .query_row(args, |row| {
+                let block_id: StacksBlockId = row.get(0)?;
+
+                Ok(block_id)
+            })
+            .optional()?)
+    }
+
+    /// Get the block ID of a staging block from block height.
+    /// Used for downloads
+    pub fn get_block_id_by_block_height(
```

Comment:
Flagging for your attention -- this is not fork-aware.  You'll need to use `IndexDBConn::get_ancestor_block_hash`.  You can get an `IndexDBConn` from `StacksChainState::index_conn()`

Reply:
i have changed the logic in the blockbyheight call to use the marf and the tip, so i can definitely remove it, thanks